apps/studio/src/server/modules/permissions/permissions.service.ts

```diff
@@ -109,3 +110,50 @@ export const validateUserPermissionsForResource = async ({
     })
   }
 }
+
+const definePermissionsForManagingUsers = async ({
+  userId,
+  siteId,
+}: Omit<PermissionsProps, "resourceId">) => {
+  const builder = new AbilityBuilder<UserManagementAbility>(createMongoAbility)
+  const roles = await db
+    .selectFrom("ResourcePermission")
+    .where("userId", "=", userId)
+    .where("siteId", "=", siteId)
+    .where("resourceId", "is", null)
+    .where("deletedAt", "is", null)
+    .select("role")
+    .execute()
+
+  // NOTE: Any role should be able to read the list of user permissions
+  if (roles.length > 0) {
```

Comment:
technically length of roles should be just 1 right? >1 would be an error state

Reply:
yes you are right, but currently we don't have constaints on the DB layer to check for uniqueness, because we are using `@@unique([userId, siteId, resourceId, role])`, and a `NULL` `resourceId` is not "unique" per se due to how DB compare and handles `NULL`

thus is more defensive to make sure any error-nous entries don't throw any error

Reply:
fixed in https://github.com/opengovsg/isomer/pull/1118/commits/f2bec85e5bb24efca8ee4d4badcbc9e9a1db7f83